packages/redux/src/raffles/selectors.ts

```diff
@@ -0,0 +1,280 @@
+import { createSelector } from 'reselect';
+import { buildQueryStringFromObject as generateRafflesHash } from '../helpers';
+import {
+  getAllRaffles as getAllRafflesFromReducer,
+  getEstimations,
+  getParticipationCreations,
+  getParticipations,
+  getRaffles as getRafflesFromReducer,
+} from './reducer';
+import { getEntities, getEntityById } from '../entities/selectors';
+import type {
+  Raffle,
+  RaffleParticipation,
+  RafflesQuery,
+} from '@farfetch/blackout-client';
+import type { RafflesState } from './types';
+import type { StoreState } from '../types/storeState.types';
+
+const getRaffleEntities = (state: StoreState) => getEntities(state, 'raffles');
+const getRaffleParticipationEntities = (state: StoreState) =>
+  getEntities(state, 'raffleParticipations');
+
+const getRaffleParticipations = createSelector(
+  [getRaffleParticipationEntities],
+  raffleParticipations => Object.values(raffleParticipations || {}),
+);
+
+/**
+ * Returns all raffles from state.
+ *
+ * @param state - Application state
+ *
+ * @returns  All Raffle entities in the state.
+ */
+export const getAllRaffles = createSelector([getRaffleEntities], raffles =>
```

Comment:
The types of the functions inferred by `createSelector` are not very user-friendly. This is one of the cases where it makes sense to type both the input and return types in typescript. Please, take a look at all the selectors in this file that use `createSelector` and type them accordingly.

Reply:
something like this? 

the type: 
```
export type DetailedRaffle = Raffle & {
  estimation: RaffleEstimation | undefined;
  participation: RaffleParticipation | undefined;
};
```

The selector
```
/**
 * Return a raffle with estimation and participation details.
 *
 * @param state - Application state.
 * @param raffleId - Raffle id.
 *
 */
export const getDetailedRaffle: (
  state: StoreState,
  raffleId: Raffle['id'],
) => DetailedRaffle | undefined = createSelector(
  [
    (state: StoreState, raffleId: Raffle['id']) => getRaffle(state, raffleId),
    (state: StoreState, raffleId: Raffle['id']) =>
      getRaffleEstimation(state, raffleId),
    (state: StoreState, raffleId: Raffle['id']) =>
      getRaffleParticipationForRaffle(state, raffleId),
  ],
  (raffle, estimation, participation) => {
    if (!raffle) {
      return;
    }
    return {
      ...raffle,
      estimation,
      participation,
    };
  },
);
```

Reply:
pls double check

Reply:
Yes, exactly. There is a problem with the `getAllRaffles` though, I will open another comment for it.